src/Silverback.Integration/Diagnostics/IBrokerHeadersEnricher.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using Silverback.Messaging.Messages;
+
+namespace Silverback.Diagnostics
+{
+    /// <summary>
+    ///     Provides custom message headers.
+    /// </summary>
+    public interface IBrokerHeadersEnricher
```

Comment:
As said in another comment, this should be reworked and we should probably reuse the more generic `IOutboundMessageEnricher` that will allow to modify more than only the headers.

Reply:
Moved to `IMovePolicyMessageEnricher`.